pythran/pythonic/numpy/fft/c2c.hpp

```diff
@@ -0,0 +1,122 @@
+#ifndef PYTHONIC_NUMPY_FFT_C2C_HPP
+#define PYTHONIC_NUMPY_FFT_C2C_HPP
+
+#include "pythonic/include/numpy/fft/c2c.hpp"
+#include "pythonic/utils/functor.hpp"
+#include "pythonic/include/utils/array_helper.hpp"
+#include "pythonic/types/ndarray.hpp"
+#include "pythonic/builtins/None.hpp"
+#include "pythonic/numpy/concatenate.hpp"
+#include "pythonic/numpy/zeros.hpp"
+
+
+#include <array>
+#include <cstring>
+#include <cmath>
+
+#include "pythonic/numpy/fft/pocketfft.hpp"
+
+PYTHONIC_NS_BEGIN
+
+namespace numpy
+{
+  namespace fft
+  {
+    using pocketfft::stride_t;
+    using pocketfft::shape_t;
+    using ldbl_t = typename std::conditional<sizeof(long double)==sizeof(double), double, long double>::type;
+
+    template<typename T> T norm_fct(int inorm, size_t N)
+    {
+        if (inorm==0) return T(1);
+        if (inorm==2) return T(1/ldbl_t(N));
+        if (inorm==1) return T(1/sqrt(ldbl_t(N)));
+            throw std::invalid_argument("invalid value for inorm (must be 0, 1, or 2)");
+    }
+
+    template<typename T> T norm_fct(int inorm, const shape_t &shape,
+    const shape_t &axes, size_t fct=1, int delta=0)
+    {
+    if (inorm==0) return T(1);
+    size_t N(1);
+    for (auto a: axes)
+        N *= fct * size_t(int64_t(shape[a])+delta);
+    return norm_fct<T>(inorm, N);
+    }
+
+    template <class T , class pS>
+    stride_t create_strides(types::ndarray<T, pS> const &in_array)
+    {
+        long npts = in_array.template shape<std::tuple_size<pS>::value - 1>();
+        auto constexpr N = std::tuple_size<pS>::value;
+        auto shape = sutils::getshape(in_array);
+        stride_t strides = stride_t(N);
+        strides[N-1] = sizeof(T);
+        std::transform(strides.rbegin(), strides.rend() - 1, shape.rbegin(),
+                  strides.rbegin() + 1, std::multiplies<long>());
+        return strides;
+    }
+
+    template <class T , class pS>
+    types::ndarray<std::complex<T>, types::array<long, std::tuple_size<pS>::value>>
+    c2c(types::ndarray<std::complex<T>, pS> const &in_array, long n, long axis,
+          types::str const &norm, bool forward)
+    {
+      long i;
+      auto constexpr N = std::tuple_size<pS>::value;
+      int inorm;
+      if (norm == "ortho") {
+        inorm = 1;
+      } else if (!forward){
+        inorm = 2;
+      }else {
+        inorm = 0;
+      }
```

Comment:
based on that code, it can even be an `enum class` which makes sure that the exception above cannot happen.

Reply:
Fixed, although I'm not sure I'm using enum classes correct (never used it before).